packages/documents/lib/resolve.js

```diff
@@ -97,28 +102,52 @@ const createUrlReplacer = (allDocuments = [], usernames = [], errors = [], urlPr
     }
   }
 
-  const repoId = getRepoId(url, 'autoSlug')
+  const { repoId, parsedUrl } = getRepoId(url, 'autoSlug')
+  // {url} lacks {repoId} and is nothing to resolve further.
   if (!repoId) {
     return url
   }
+
+  // Return early if {stripDocLinks} argument is set as nothing has to be resolved.
   if (stripDocLinks) {
     return ''
   }
+
   const linkedDoc = allDocuments
     .find(d => d.meta.repoId === repoId)
+
   if (linkedDoc) {
-    const hash = url.split('#')[1]
-    return `${urlPrefix}${linkedDoc.content.meta.path}${searchString}${hash ? `#${hash}` : ''}`
+    // Stitch and parse simple URL version including arguments {urlPrefix}, {searchString}
+    const resolvedUrl = new URL(
+      `${urlPrefix}${linkedDoc.content.meta.path}${searchString}`,
+      FRONTEND_BASE_URL
```

Comment:
`FRONTEND_BASE_URL` wasn't involved here before. Are you sure it doesn't have any unwanted consequences?

Reply:
Yes, as sure as automated and manual testing made me.